indra/literature/pubmed_client.py

```diff
@@ -307,12 +307,130 @@ def _find_elem_text(root, xpath_string):
+    issue = journal.find('JournalIssue')
+    issue_volume = _find_elem_text(issue, 'Volume')
+    issue_issue = _find_elem_text(issue, 'Issue')
+
+    issue_pub_date = issue.find('PubDate')
+    if issue_pub_date is not None:
+        # Get issue year
+        issue_year = _find_elem_text(issue_pub_date, 'Year')
+        issue_year = int(issue_year) if issue_year else None
+
+    else:
+        issue_year = None
+
+    return {
+        "issue_volume": issue_volume,
+        "issue_issue": issue_issue,
+        "issue_year": issue_year
+    }
+
+
+def get_issn_info(
+    medline_citation: ET.Element,
+    get_issns_from_nlm: Literal["never", "missing", "always"] = "never"
+):
+    """Given a medline citation, get the issn info from the article
+
+    Parameters
+    ----------
+    medline_citation : xml.etree.ElementTree.Element
+        The MedlineCitation element of the PubMed XML tree.
+    get_issns_from_nlm : Literal['never', 'missing', 'always']
+        Whether to recover ISSN values from the NLM catalog. Options are
+        'never', 'missing', and 'always'. If 'missing', then the ISSN
+        values will be recovered from the NLM catalog if they are not found
+        in the XML. If 'always', then the ISSN values will be recovered from
+        the NLM catalog regardless of whether they are found in the XML.
+        Default is 'never' (i.e., never recover from NLM catalog regardless
+        of whether they are found in the XML).
+
+    Returns
+    -------
+    dict
+        A dictionary journal, issue, and ISSN info. The structure is as
+        follows:
+        {
+            "journal_title": str,
+            "journal_abbrev": str,
+            "journal_nlm_id": str,
+            "issn_dict": {
+                "issn": str,
+                "issn_l": str,
+                "p_issn": str,
+                "e_issn": str
+                "other": [str, ...]
+            },
+            "issue_dict": {
+                "issue_volume": str,
+                "issue": str,
+                "issue_year": int
+            }
+        }
+    """
+    if get_issns_from_nlm not in ['never', 'missing', 'always']:
+        raise ValueError("get_issns_from_nlm must be one of 'never', "
+                         "'missing', or 'always'")
     # Journal info
     journal = medline_citation.find('Article/Journal')
     journal_title = _find_elem_text(journal, 'Title')
     journal_abbrev = _find_elem_text(journal, 'ISOAbbreviation')
 
+    # Issue info
+    issue_info = _get_issue_info(journal)
+
+    # Get the ISSN from the article record
+    issn_dict = {}
+    issn_element = journal.find("ISSN")
+    if issn_element is not None:
+        issn_type = issn_element.attrib["IssnType"]
+        issn = issn_element.text
+        issn_dict["issn"] = issn
+        if issn_type.lower() in ["electronic", "print"]:
+            type_prefix = "p_" if issn_type.lower() == "print" else "e_"
+            issn_dict[type_prefix + "issn"] = issn
```

Comment:
why not just create a data structure that captures this in a structured way, rather than adding cryptic keys? Is it possible there are multiple print/electronic, or even other types? I definitely saw this scenario on wikidata. Instead, you could make a list like
```
[
  { "issn": XXX, "type": "print" },
  { "issn": YYY, "type": "electronic" },
  { "issn": ZZZ, "type": "other" },
]
``

Reply:
Per journal you typically only have the ISSN (whether it be 'print', 'electronic' or other categories) and the ISSN-L (none of them are guaranteed to be present though, but the NLM ID is as far as I've seen), so I don't think we need to do a list, but I agree that the structure can be simplified.